sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -25,48 +25,49 @@
 public final class ActivityFramesTracker {
 
   private @Nullable FrameMetricsAggregator frameMetricsAggregator = null;
-  private boolean androidXAvailable = true;
+  private final SentryAndroidOptions options;
 
   private final @NotNull Map<SentryId, Map<String, @NotNull MeasurementValue>>
       activityMeasurements = new ConcurrentHashMap<>();
   private final @NotNull Map<Activity, FrameCounts> frameCountAtStartSnapshots =
       new WeakHashMap<>();
 
-  private final @Nullable ILogger logger;
   private final @NotNull MainLooperHandler handler;
 
   public ActivityFramesTracker(
       final @NotNull LoadClass loadClass,
-      final @Nullable ILogger logger,
+      final @NotNull SentryAndroidOptions options,
       final @NotNull MainLooperHandler handler) {
-    androidXAvailable =
-        loadClass.isClassAvailable("androidx.core.app.FrameMetricsAggregator", logger);
+
+    final boolean androidXAvailable =
+        loadClass.isClassAvailable("androidx.core.app.FrameMetricsAggregator", options.getLogger());
+
     if (androidXAvailable) {
       frameMetricsAggregator = new FrameMetricsAggregator();
     }
-    this.logger = logger;
+    this.options = options;
     this.handler = handler;
   }
 
-  public ActivityFramesTracker(final @NotNull LoadClass loadClass, final @Nullable ILogger logger) {
-    this(loadClass, logger, new MainLooperHandler());
-  }
-
-  public ActivityFramesTracker(final @NotNull LoadClass loadClass) {
```

Comment:
I think we should keep this ctor because flutter is using it https://github.com/getsentry/sentry-dart/blob/5d15a51e3c8c0d5af29d48868ee07f994610b274/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt#L157-L162

Or is it fine to also change it on Flutter and pass the options there @marandaneto ?

Reply:
If that fixes or improves something on the frames tracker, I can fix Flutter as well.
Raise an issue on Flutter and likely RN as well, so we know what we have to fix exactly (didn't look into the details).

Reply:
Well spotted! It doesn't improve anything, but gives you the possibility to opt-out of `ActivityFramesTracker`. I can add the old constructor back-in to avoid anything breaking + create a dart PR for the new constructor.

Reply:
actually, I just noticed that the original issue was about flutter (the customer is using our flutter sdk), so we'd need this on Flutter as well, to have it fixed for them.

@markushi sounds good re. the old constructor and the PR. 👍

Reply:
Sounds good, I added the old constructor and created a simple draft PR in the sentry-dart repo: https://github.com/getsentry/sentry-dart/pull/1087